mesonbuild/backend/ninjabackend.py

```diff
@@ -1985,6 +1985,9 @@ def generate_rust_target(self, target: build.BuildTarget) -> None:
         else:
             verbatim = ''
 
+        if cratetype in {'dylib', 'proc-macro', 'plugin'}:
```

Comment:
plugin is not a valid crate type I think

Reply:
Ah, they're deprecated or something.  I'll remove plugin.

Reply:
The only valid ones are dylib, cdylib, rlib, staticlib, proc-macro, and special lib and clib AFAIK CC @xavierclaessens

Reply:
(the special ones are translated to the normal ones in `build.py` so are irrelevant here)